Components/ProductToShop.php

```diff
@@ -435,6 +442,78 @@ public function insertOrUpdate(Product $product)
     }
 
     /**
+     * @param ProductModel $article
+     * @param Product $product
+     */
+    private function applyProductProperties(ProductModel $article, Product $product)
+    {
+        if (empty($product->properties)) {
+            return;
+        }
+
+        /** @var Property $firstProperty */
+        $firstProperty = reset($product->properties);
+        $groupRepo = $this->manager->getRepository(PropertyGroup::class);
+        $group = $groupRepo->findOneBy(['name' => $firstProperty->groupName]);
+
+        if (!$group) {
+            $group = new PropertyGroup();
+            $group->setName($firstProperty->groupName);
+            $group->setComparable($firstProperty->comparable);
+            $group->setSortMode($firstProperty->sortMode);
+            $group->setPosition($firstProperty->groupPosition);
+        }
+
+        $article->setPropertyGroup($group);
+
+        $valueCollection = [];
+        $optionRepo = $this->manager->getRepository(PropertyOption::class);
+        $valueRepo = $this->manager->getRepository(PropertyValue::class);
+
+        foreach ($product->properties as $property) {
+
+            $option = $optionRepo->findOneBy(['name' => $property->option]);
+
+            if (!$option) {
+                $option = new PropertyOption();
+                $option->setName($property->option);
+                $option->setFilterable($property->filterable);
+                $this->manager->persist($option);
+                $this->manager->flush($option);
+            }
+
+            $value = $valueRepo->findOneBy(['optionId' => $option->getId(), 'value' => $property->value]);
+
+            if (!$value) {
+                $value = new PropertyValue($option, $property->value);
+                $value->setPosition($property->valuePosition);
+                $this->manager->persist($value);
+            }
+
+            $valueCollection[] = $value;
+
+            $filters = [
+                ['property' => "options.name", 'expression' => '=', 'value' => $property->option],
+                ['property' => "groups.name", 'expression' => '=', 'value' => $property->groupName],
+            ];
+
+            $query = $groupRepo->getPropertyRelationQuery($filters, null, 1, 0);
+            $relation = $query->getOneOrNullResult();
+
+            if (!$relation) {
+                $group->addOption($option);
+                $this->manager->persist($group);
+                $this->manager->flush($group);
```

Comment:
what about if remove this and use last flush without arguments at the end of this function?